File: scripts/clip.py
import subprocess
import sys
import os

GREEN_TEXT = "\033[92m"
RESET_TEXT = "\033[0m"
BLUE_TEXT = "\033[34m"


def copy_to_clipboard(file_path):
    try:
        with open(file_path, "rb") as file:
            process = subprocess.Popen("pbcopy", stdin=subprocess.PIPE)
            process.communicate(file.read())

        file_name = os.path.basename(file_path)

        print(
            f"{GREEN_TEXT}Copied{RESET_TEXT} {BLUE_TEXT}{file_name}{RESET_TEXT} {GREEN_TEXT}to clipboard{RESET_TEXT}"
        )

    except FileNotFoundError:
        print(f"{BLUE_TEXT}{file_path}{RESET_TEXT} not found")
    except Exception as e:
        print(f"An error occurred: {e}")


if __name__ == "__main__":
    if len(sys.argv) != 2:
        print(f"Usage: python clip.py {BLUE_TEXT}<file_path>{RESET_TEXT}")
    else:
        copy_to_clipboard(sys.argv[1])

File: scripts/go/clip/clip.go
package main

import (
	"fmt"
	"io"
	"os"
	"os/exec"
	"path/filepath"
)

func copyToClipboard(filePath string) error {
	// Open the file
	file, err := os.Open(filePath)
	if err != nil {
		return fmt.Errorf("failed to open file: %v", err)
	}
	defer file.Close()

	// Create pbcopy command
	cmd := exec.Command("pbcopy")
	stdin, err := cmd.StdinPipe()
	if err != nil {
		return fmt.Errorf("failed to create stdin pipe: %v", err)
	}

	// Start the command
	if err := cmd.Start(); err != nil {
		return fmt.Errorf("failed to start pbcopy: %v", err)
	}

	// Copy file content to pbcopy's stdin
	if _, err := io.Copy(stdin, file); err != nil {
		return fmt.Errorf("failed to copy to clipboard: %v", err)
	}

	// Close stdin and wait for pbcopy to finish
	stdin.Close()
	if err := cmd.Wait(); err != nil {
		return fmt.Errorf("pbcopy failed: %v", err)
	}

	fileName := filepath.Base(filePath)
	fmt.Printf("Copied %s to clipboard\n", fileName)
	return nil
}

func main() {
	if len(os.Args) != 2 {
		fmt.Println("Usage: clip <file_path>")
		os.Exit(1)
	}

	if err := copyToClipboard(os.Args[1]); err != nil {
		fmt.Println("Error:", err)
		os.Exit(1)
	}
}
